plugins/destination/gremlin/client/spec.go

```diff
@@ -91,3 +109,108 @@ func (s *Spec) Validate() error {
 
 	return nil
 }
+
+func (Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
+	authMode := sc.Properties.Value("auth_mode")
+	authMode.Enum = []any{authModeNone, authModeBasic, authModeAWS}
+
+	sc.OneOf = []*jsonschema.Schema{
+		{
+			// if auth_mode is aws, require aws_region to be set
+			Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+				one := uint64(1)
+				properties := jsonschema.NewProperties()
+
+				authMode := *sc.Properties.Value("auth_mode")
+				authMode.Enum = []any{authModeAWS}
```

Comment:
instead of [`enum`](https://json-schema.org/draft/2020-12/json-schema-validation#section-6.1.2) use [`const`](https://json-schema.org/draft/2020-12/json-schema-validation#section-6.1.3)